test/recursive_array_tools.jl

```diff
@@ -0,0 +1,12 @@
+using RecursiveArrayTools
+
+function f!(du, u)
+    du.foo[2] = u.foo[1]
+    du.foo[1] = u.foo[2]
+    return nothing
+end
+
+u = NamedArrayPartition(; foo=rand(5))
+du = copy(u)
+
+jacobian_sparsity(f!, du, u, TracerSparsityDetector())
```

Comment:
Would be nice to have a slightly more rigorous test than just a dry-run. 

I don't know the functionality of `NamedArrayPartition`, but the tests should cover as much of it as possible.